apps/console/src/features/applications/components/application-list.tsx

```diff
@@ -543,7 +545,12 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
                             setShowDeleteConfirmationModal(false);
                             setAlert(null);
                         } }
-                        onPrimaryActionClick={ (): void => handleApplicationDelete(deletingApplication.id) }
+                        onPrimaryActionClick={ (): void => {
+                            setLoading(true);
+                            handleApplicationDelete(deletingApplication.id).finally(() => {
```

Comment:
We can extract this callback's inner function calls (setLoading and .finally() part) and place them inside **handleApplicationDelete** function. So we can have containerised handleApplicationDelete function, which can be used in other places too without duplicating the logic of handling setLoading, thus can remove the promise return from the **handleApplicationDelete** function as well

Reply:
Addressed via af36e121eed12e6c0f06003fee7e01072b7c4ee6